src/main/java/org/elasticsearch/plugin/zentity/ResolutionAction.java

```diff
@@ -1,30 +1,71 @@
 package org.elasticsearch.plugin.zentity;
 
+import io.zentity.common.AsyncCollectionRunner;
+import io.zentity.common.Json;
+import io.zentity.common.StreamUtil;
 import io.zentity.model.Model;
-import io.zentity.model.ValidationException;
 import io.zentity.resolution.Job;
 import io.zentity.resolution.input.Input;
+import joptsimple.internal.Strings;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionListener;
-import org.elasticsearch.action.get.GetResponse;
 import org.elasticsearch.client.node.NodeClient;
-import org.elasticsearch.rest.*;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestStatus;
 
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
 
+import static java.util.Collections.emptyMap;
 import static org.elasticsearch.rest.RestRequest.Method.POST;
 
 
 public class ResolutionAction extends BaseRestHandler {
 
     private static final Logger logger = LogManager.getLogger(ResolutionAction.class);
 
+    private static final int MAX_CONCURRENT_JOBS_PER_REQUEST = 100;

```

Comment:
I agree with that analysis and conclusion about tieing it to the NodeClient's client values. I think it might make sense to expose this setting to the user at some point, as it is likely a deployment-specific parameter for tuning.